### PR TITLE
Add ghc 9.6.1 to the matrix

### PR DESCRIPTION
**Pull request by angerman**

### CODE REVIEW

**Review by bolt12 — APPROVED** [review]

File: io-sim/src/Data/Deque/Strict.hs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+import           Prelude hiding (head, init, tail)
+import           Data.Foldable (foldl')
+import qualified Data.List as List
+
+data Deque a = Deque ![a] ![a]
+
+instance Semigroup (Deque a) where
+    Deque as bs <> Deque as' bs' =
+      Deque as (bs ++ reverse as' ++ bs')
```
Comment (bolt12) [inline io-sim/src/Data/Deque/Strict.hs:15]: Shouldn't this be

```suggestion
      Deque as (bs' ++ reverse as' ++ bs)
```

File: io-sim/src/Data/Deque/Strict.hs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+instance Monoid (Deque a) where
+    mempty = Deque [] []
+
+instance Foldable Deque where
+    foldr step init (Deque head tail) =
+      foldr step (foldr step init (reverse tail)) head
```
Comment (bolt12) [inline io-sim/src/Data/Deque/Strict.hs:22]: We can avoid reversing here with `foldl' (flip step)`

File: io-sim/src/Data/Deque/Strict.hs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    foldl' step init (Deque head tail) =
+      foldl' step (foldl' step init head) (reverse tail)
```
Comment (bolt12) [inline io-sim/src/Data/Deque/Strict.hs:24]: Same here with `foldr' (flip step)`

**Review by bolt12 — CHANGES_REQUESTED** [review]
I think I found some things that need fixing

File: io-sim/src/Data/Deque/Strict.hs (hunk 1 above)
Comment (coot) [reply io-sim/src/Data/Deque/Strict.hs:15]: You're right, I messed this up after trying to do that on the head side.

File: io-sim/src/Data/Deque/Strict.hs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      Deque as (bs' ++ reverse as' ++ bs)
+
+instance Monoid (Deque a) where
+    mempty = Deque [] []
+
+instance Foldable Deque where
+    foldr step init (Deque head tail) =
+      foldr step (foldl' (flip step) init tail) head
+    foldl' step init (Deque head tail) =
+      foldr (flip step) (foldl' step init head) tail
```
Comment (bolt12) [inline io-sim/src/Data/Deque/Strict.hs:24]: I think this should be `foldr'` since we are implementing `foldl'`

**Review by bolt12 — APPROVED** [review]
Just one small thing, I'll approve in the meantime :)

**Review by bolt12 — APPROVED** [review]

**Pull request merged**